gen/src/write.rs

```diff
@@ -46,6 +46,7 @@ pub(super) fn gen(
         }
     }
 
+    let mut cxx_types = HashSet::new();
```

Comment:
The set of extern C++ type names is already available in `types.cxx`. Does that existing set work for what we need here?